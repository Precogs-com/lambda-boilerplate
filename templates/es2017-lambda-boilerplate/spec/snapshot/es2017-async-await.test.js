const { transform } = require('./utils');

it('async named function', () => {
    const code = transform(`
async function f() {
    return true;
}
`);
    expect(code).toMatchSnapshot();
});

it('async function expression', () => {
    const code = transform(`
var f = async function() {
    return true;
}
`);
    expect(code).toMatchSnapshot();
});

it('async arrow function', () => {
    const code = transform(`
var f = async () => {
    return true;
}
`);
    expect(code).toMatchSnapshot();
});

it('return await primitive', () => {
    const code = transform(`
async function f() {
    return await 123;
}
`);
    expect(code).toMatchSnapshot();
});

it('return await function call', () => {
    const code = transform(`
async function f() {
    return await g();
}
`);
    expect(code).toMatchSnapshot();
});

it('return await Promise', () => {
    const code = transform(`
async function f() {
    return await Promise.resolve(1);
}
`);
    expect(code).toMatchSnapshot();
});

it('variable assignment await primitive', () => {
    const code = transform(`
async function f() {
    var x = await 123;
    return x;
}
`);
    expect(code).toMatchSnapshot();
});

it('variable assignment await function call', () => {
    const code = transform(`
async function f() {
    var x = await g();
    return x;
}
`);
    expect(code).toMatchSnapshot();
});

it('variable assignment await Promise', () => {
    const code = transform(`
async function f() {
    var x = await Promise.resolve(1);
    return x;
}
`);
    expect(code).toMatchSnapshot();
});
