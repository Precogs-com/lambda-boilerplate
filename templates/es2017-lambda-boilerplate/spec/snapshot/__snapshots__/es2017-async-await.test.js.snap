// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async arrow function 1`] = `
"\\"use strict\\";

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step(\\"next\\", value); }, function (err) { step(\\"throw\\", err); }); } } return step(\\"next\\"); }); }; }

var f = (() => {
    var _ref = _asyncToGenerator(function* () {
        return true;
    });

    return function f() {
        return _ref.apply(this, arguments);
    };
})();"
`;

exports[`async function expression 1`] = `
"\\"use strict\\";

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step(\\"next\\", value); }, function (err) { step(\\"throw\\", err); }); } } return step(\\"next\\"); }); }; }

var f = (() => {
    var _ref = _asyncToGenerator(function* () {
        return true;
    });

    return function f() {
        return _ref.apply(this, arguments);
    };
})();"
`;

exports[`async named function 1`] = `
"\\"use strict\\";

let f = (() => {
    var _ref = _asyncToGenerator(function* () {
        return true;
    });

    return function f() {
        return _ref.apply(this, arguments);
    };
})();

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step(\\"next\\", value); }, function (err) { step(\\"throw\\", err); }); } } return step(\\"next\\"); }); }; }"
`;

exports[`return await Promise 1`] = `
"\\"use strict\\";

let f = (() => {
    var _ref = _asyncToGenerator(function* () {
        return yield Promise.resolve(1);
    });

    return function f() {
        return _ref.apply(this, arguments);
    };
})();

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step(\\"next\\", value); }, function (err) { step(\\"throw\\", err); }); } } return step(\\"next\\"); }); }; }"
`;

exports[`return await function call 1`] = `
"\\"use strict\\";

let f = (() => {
    var _ref = _asyncToGenerator(function* () {
        return yield g();
    });

    return function f() {
        return _ref.apply(this, arguments);
    };
})();

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step(\\"next\\", value); }, function (err) { step(\\"throw\\", err); }); } } return step(\\"next\\"); }); }; }"
`;

exports[`return await primitive 1`] = `
"\\"use strict\\";

let f = (() => {
    var _ref = _asyncToGenerator(function* () {
        return yield 123;
    });

    return function f() {
        return _ref.apply(this, arguments);
    };
})();

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step(\\"next\\", value); }, function (err) { step(\\"throw\\", err); }); } } return step(\\"next\\"); }); }; }"
`;

exports[`variable assignment await Promise 1`] = `
"\\"use strict\\";

let f = (() => {
    var _ref = _asyncToGenerator(function* () {
        var x = yield Promise.resolve(1);
        return x;
    });

    return function f() {
        return _ref.apply(this, arguments);
    };
})();

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step(\\"next\\", value); }, function (err) { step(\\"throw\\", err); }); } } return step(\\"next\\"); }); }; }"
`;

exports[`variable assignment await function call 1`] = `
"\\"use strict\\";

let f = (() => {
    var _ref = _asyncToGenerator(function* () {
        var x = yield g();
        return x;
    });

    return function f() {
        return _ref.apply(this, arguments);
    };
})();

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step(\\"next\\", value); }, function (err) { step(\\"throw\\", err); }); } } return step(\\"next\\"); }); }; }"
`;

exports[`variable assignment await primitive 1`] = `
"\\"use strict\\";

let f = (() => {
    var _ref = _asyncToGenerator(function* () {
        var x = yield 123;
        return x;
    });

    return function f() {
        return _ref.apply(this, arguments);
    };
})();

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step(\\"next\\", value); }, function (err) { step(\\"throw\\", err); }); } } return step(\\"next\\"); }); }; }"
`;
